### PR TITLE
CODE RUB: Standard Version `v2.10.1` Upgrade

### PR DESCRIPTION
**Pull request by aosunlana**
Closes: #274

### CODE REVIEW

**Review by glhays — COMMENTED** [review]
Set Ln(179) to private in Foundations.md

**Review by glhays — COMMENTED** [review]
Fix indenting LN(389-402) The entire block needs to be aligned left.

**Review by glhays — COMMENTED** [review]
Ln(517) make private

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 1]
```diff
@@ -686,10 +694,13 @@ private async ValueTask<Student> TryCatch(ReturningStudentFunction returningStud
 
 private StudentValidationException CreateAndLogValidationException(Exception exception)
```
Comment (glhays) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:695]: Should we be passing an Xeption here.

**Review by glhays — COMMENTED** [review]
Ln(729) make private

**Review by glhays — COMMENTED** [review]
Ln(857) same

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 2]
```diff
@@ -932,23 +940,26 @@ This implementation will take care of detecting an issue and issuing a local exc
...
+    {
+        throw CreateAndLogValidationException(notFoundStudentException);
+    }
 }
 
 private StudentValidationException CreateAndLogValidationException(Exception exception)
 {
-	var studentValidationException = new StudentValidationException(exception);
-	this.loggingBroker.LogError(studentValidationException);
+    var studentValidationException = new StudentValidationException(
```
Comment (glhays) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:956]: Xeption?

**Review by glhays — COMMENTED** [review]
Ln(1004) same

**Review by glhays — COMMENTED** [review]
What about sections 2.3 and 2.4? All we have here is section 2.1
@aosunlana I believe there may be some changes in these other sections.

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 3]
```diff
@@ -589,16 +585,27 @@ Our `StudentValidationException` would be implemented as follows:
 ```csharp
 public class StudentValidationException : Exception
 {
-	public StudentValidationException(Exception innerException)
-		: base("Student validation error occurred, please check your input and try again.",
-			innerException) { }
+    public StudentValidationException(string message, Exception innerException)
```
Comment (cjdutoit) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:588]:
```suggestion
    public StudentValidationException(string message, Xeption innerException)
```

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 4]
```diff
@@ -589,16 +585,27 @@ Our `StudentValidationException` would be implemented as follows:
...
+        : base(message, innerException)
+    { }
+}
+```
+The string messaging for the outer-validation above will be passed when the exception is initialized from the service class as shown below.
+
+```csharp
+private StudentValidationException CreateAndLogValidationException(Exception exception)
```
Comment (cjdutoit) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:596]:
```suggestion
private StudentValidationException CreateAndLogValidationException(Xeption exception)
```

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 5]
```diff
@@ -932,23 +940,26 @@ This implementation will take care of detecting an issue and issuing a local exc
...
-	catch (NotFoundStudentException notFoundStudentException)
-	{
-		throw CreateAndLogValidationException(notFoundStudentException);
-	}
+    try
+    {
+        return await returningStudentFunction();
+    }
+    ..
+        catch (NotFoundStudentException notFoundStudentException)
```
Comment (cjdutoit) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:948]:
```suggestion
    catch (NotFoundStudentException notFoundStudentException)
```

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 6]
```diff
@@ -1089,13 +1097,9 @@ We also need the `StudentDependencyValidationException` which should be as follo
 ```csharp
 public class StudentDependencyValidationException : Exception
 {
-	public StudentDependencyValidationException(Exception innerException)
-		: base("Student dependency validation error occurred, please try again.", innerException){ }
-
-	public StudentDependencyValidationException(string message, Exception innerException)
-		: base (message, innerException) {}
+    public StudentDependencyValidationException(string message, Exception innerException)
```
Comment (cjdutoit) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:1100]:
```suggestion
    public StudentDependencyValidationException(string message, Xeption innerException)
```

File: 2. Services/2.1 Foundations/2.1 Foundations.md [hunk 7]
```diff
@@ -1107,30 +1111,34 @@ private delegate ValueTask<Student> ReturningStudentFunction();
...
-			new AlreadyExistsStudentException(duplicateKeyException);
-
-		throw CreateAndLogDependencyValidationException(alreadyExistsStudentException);
-	}
+    try
+    {
+        return await returningStudentFunction();
+    }
+    ...
+        catch (DuplicateKeyException duplicateKeyException)
```
Comment (cjdutoit) [inline 2. Services/2.1 Foundations/2.1 Foundations.md:1119]:
```suggestion
    catch (DuplicateKeyException duplicateKeyException)
```

**Comment by hassanhabib** [comment]
@glhays @cjdutoit I'm assuming this is still pending right?

**Comment by glhays** [comment]
> @glhays @cjdutoit I'm assuming this is still pending right?

@hassanhabib Yes waiting for @aosunlana to resolve comments.

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 4 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:596]: Treated

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 1 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:695]: Treated

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 5 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:948]: Treated

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 2 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:956]: Treated

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 6 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:1100]: Treated

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 7 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:1119]: Treated

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 3 above)
Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:588]: Treated

**Comment by aosunlana** [comment]
> > @glhays @cjdutoit I'm assuming this is still pending right?
> 
> @hassanhabib Yes waiting for @aosunlana to resolve comments.

@glhays @cjdutoit @hassanhabib all comments resolved and ready for review.

File: 2. Services/2.1 Foundations/2.1 Foundations.md (hunk 3 above)
Comment (cjdutoit) [reply 2. Services/2.1 Foundations/2.1 Foundations.md:588]: This still needs to change to Xeption.

Comment (aosunlana) [reply 2. Services/2.1 Foundations/2.1 Foundations.md]: Treated

File: 2. Services/2.3 Orchestrations/2.3 Orchestrations.md [hunk 8]
```diff
@@ -740,7 +740,9 @@ public partial class StudentOrchestrationService
     private StudentOrchestrationDependencyValidationException CreateAndLogDependencyValidationException(Xeption exception)
     {
         var studentOrchestrationDependencyValidationException =
-            new StudentOrchestrationDependencyValidationException(exception.innerException as Xeption);
+            new StudentOrchestrationDependencyValidationException(
+                message: "Student dependency validation error occurred, fix errors and try again,
```
[the remark this answers is not in the document]
Comment (aosunlana) [reply 2. Services/2.3 Orchestrations/2.3 Orchestrations.md:744]: Treated

File: 2. Services/2.3 Orchestrations/2.3 Orchestrations.md [hunk 9]
```diff
@@ -740,7 +740,9 @@ public partial class StudentOrchestrationService
...
+                exception.innerException as Xeption);
```
[the remark this answers is not in the document]
Comment (aosunlana) [reply 2. Services/2.3 Orchestrations/2.3 Orchestrations.md:745]: Treated

**Pull request merged**